homeassistant/components/counter/__init__.py

```diff
@@ -120,32 +143,59 @@ def state(self):
     @property
     def state_attributes(self):
         """Return the state attributes."""
-        return {
+        ret = {
             ATTR_INITIAL: self._initial,
             ATTR_STEP: self._step,
         }
+        if self._min is not None:
+            ret[CONF_MINIMUM] = self._min
+        if self._max is not None:
+            ret[CONF_MAXIMUM] = self._max
+        return ret
+
+    def __check_boundaries(self):
```

Comment:
It feels cleaner if this returns a true/false rather than making the change directly.